### PR TITLE
Comment model

### PR DESCRIPTION
**Pull request by mattegan111**

### CODE REVIEW

File: server/models/Comment.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const CommentSchema = mongoose.Schema({
+	bodytext: {
+		type: String,
+		required: true,
+	},
+
+	author: {
+		type: mongoose.Schema.Types.ObjectId,
+		required: true,
+	},
```
Comment (nathanrobb) [inline server/models/Comment.js:12]: Should this ref the user? `ref: 'User',`

File: server/models/Comment.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	post: {
+		type: mongoose.Schema.Types.ObjectId,
+		required: true,
+	},
```
Comment (nathanrobb) [inline server/models/Comment.js:17]: Should this ref the posts `ref: 'Post',`?

File: server/models/User.js [hunk 3]
```diff
@@ -37,9 +37,12 @@ const UserSchema = mongoose.Schema({
 	post: {
 		type: Array,
 	},
```
Comment (nathanrobb) [inline server/models/User.js:39]: Should this be updated in #4?

File: server/routes/comments.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+const {
+	body,
+	validationResult,
+} = require('express-validator');
+const ObjectId = require('mongoose').Types.ObjectId;
+const router = express.Router();
+
+const Comment = require('../models/Comment');
+
+// @route   GET api/comments
```
Comment (nathanrobb) [inline server/routes/comments.js:13]: When would we use this endpoint? Listing all the users comments? I feel like we'd usually just want these retrieved when we get a post?

File: server/routes/comments.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+			return res
+				.status(404)
+				.json({ msg: 'No comments found' });
+		}
+
+		res.json(comments);
+	})
+);
+
+// @route   POST api/comments
```
Comment (nathanrobb) [inline server/routes/comments.js:36]: Did you consider doing something like this as a route instead `api/posts/:postId/comments` and `PUT api/posts/:postId/comments/:commentId`

I guess it makes the delete endpoint possibly a little strange (if we didn't want to require a post id... 🤔 maybe not an issue).

File: server/routes/comments.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+			post,
+			author: req.user.id,
+		});
+
+		await comment.save();
+		res.json(comment);
+	})
+);
+
+// @route   PUT api/comments/:id
```
Comment (nathanrobb) [inline server/routes/comments.js:67]: We can only edit a comment or like a comment, we cant do both in the same request.

We should add a like endpoint IMO. eg `PATCH api/comments/:id/like` or PUT or w/e

File: server/routes/comments.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+);
+
+// @route   DELETE api/comments/:id
+// @desc    Delete comment
+// @access  Private
+router.delete(
+	'/:id',
+	auth,
+	errorCatcher(async (req, res) => {
+		const comment = await Comment.findById(req.params.id);
```
Comment (nathanrobb) [inline server/routes/comments.js:153]: We could do a where on the db here? Not sure how to do it in mongoose

```js
		const comment = await Comment.findById(req.params.id).Where({ author = req.user.id});
```

**Review by nathanrobb — CHANGES_REQUESTED** [review]
Suggest splitting the like out of the comment PUT.

Possibly something we want to do for the Post routes too.

**Comment by nathanrobb** [comment]
Hmm seems this auto closed, thought that the base branch would switch to `main`... my bad.

Looks like there an active discussion to implement what I thought would happen here. https://github.com/github/feedback/discussions/4477

EDIT: actually, perhaps this discussion is about forked repos and not related to this one 🤔

**Comment by mattegan111** [comment]
Ok cool, I'm gonna try opening another PR for the comment-model branch. (It might need a unique name like reopen-comment-model.) You might need to jump between PR's for context sorry!

Not sure yet how to handle notification mode, maybe deal with it after the main is updated with all the comment stuff and merge main into notification-model.

File: server/models/User.js (hunk 3 above)
Comment (mattegan111) [reply server/models/User.js:39]: Yes, looks like I missed this and #4 is merged now, maybe I could've read comments across all branches before merging. I think it'd be best to open another branch for this later.

Comment (mattegan111) [reply server/models/User.js:39]: Same with `friends`, `receivedfriendrequests` and `sentfriendrequests`

File: server/routes/comments.js (hunk 4 above)
Comment (mattegan111) [reply server/routes/comments.js:13]: Could be useful for a "User's Recent Activity" component. But is 'could be useful' a good enough reason to merge into main?

File: server/routes/comments.js (hunk 5 above)
Comment (mattegan111) [reply server/routes/comments.js:36]: Your suggestion seems to better follow RESTful architecture, one potential advantage is having the post id accessible through the params. Let's use it.

Delete endpoint might be ok.

Comment (mattegan111) [reply server/routes/comments.js:36]: Your suggestion is a proper RESTful architecture - I want to implement it but I realised I'd probably have to completely change the file structure for routing. It'd be nice to come back to it later if you think the current solution is good enough?

File: server/routes/comments.js (hunk 7 above)
Comment (mattegan111) [reply server/routes/comments.js:153]: Ok cool. What's the advantage? Fewer lines of code? More efficient db requests?

**Pull request closed without merging**